subsys/net/ip/net_if.c

```diff
@@ -4145,12 +4145,16 @@ enum net_verdict net_if_recv_data(struct net_if *iface, struct net_pkt *pkt)
 		/* L2 has modified the buffer starting point, it is easier
 		 * to re-initialize the cursor rather than updating it.
 		 */
-		net_pkt_cursor_init(new_pkt);
+		if (new_pkt) {
+			net_pkt_cursor_init(new_pkt);
 
-		if (net_promisc_mode_input(new_pkt) == NET_DROP) {
-			net_pkt_unref(new_pkt);
+			if (net_promisc_mode_input(new_pkt) == NET_DROP) {
+				net_pkt_unref(new_pkt);
+			}
+		} else {
+			NET_WARN("packet dropped, unable to clone packet");
+			verdict = NET_DROP;
```

Comment:
After revisiting this change, I am not sure about this `NET_DROP`. The reason being that the function should return info whether the original net_pkt should be dropped or not, and in this case the earlier call
`verdict = net_if_l2(iface)->recv(iface, pkt);` returns that information.
So for the cloned pkt, imho we should not set the verdict here. To me it looks like that if we return `NET_DROP` here, the original pkt is dropped without any reason. WDYT?

Reply:
Also in the case of the cloning failed, perhaps better error msg would be
`NET_WARN("promiscuous packet dropped, unable to clone packet");`
in order to tell the reader what exactly happened.

Reply:
After looking into it I agree with your suggestion and have removed the verdict part.